docs/content/troubleshooting/log_status_collect.md

```diff
@@ -1,41 +1,121 @@
 
 **Note:** These procedures are applicable for both Kubernetes and Red Hat® OpenShift®. For Red Hat OpenShift, replace `kubectl` with `oc` in all relevant commands.
 
-To collect and display status and logs related to the different components of IBM® block storage CSI driver, use the following Kubernetes commands and procedures:
+## Detecting errors
+
+To help pinpoint potential causes for stateful pod failure:
+
+1.  Verify that all CSI pods are running.
+    ```
+    kubectl get pods -n <namespace> -l csi
+    ```
+
+2.  If a pod is not in a _Running_ state, run the following command:
+    ```
+    kubectl describe -n <namespace> pod/<pod-name>
+    ```
+    View the logs.
+
+## Status and log collection
+To collect and display status and logs related to the different components of IBM® block storage CSI driver, use the Kubernetes commands that are found in this section.
+
+Before you begin collecting logs, create a directory for the logs on the control-plane or bastion server.
+
+For example:
+
+```
+mkdir logs
+```
+Save logs and status reports directly to the created directory by adding in the following string at the end of collection command: `> logs/<log_filename>`.
+
+**Important:** Be sure that the logs cover any relevant timeframes for the specific issues that you are trying to debug when gathering logs from the storage system.
+
+### General status and log collection for IBM Support
+Be sure to run the following steps and copy the output to an external file, when engaging IBM Support and sending log collections.
+
+1. Check the node status.
+    
+    `kubectl get nodes`
+2. Check the CSI driver component status.
+
+    `kubectl get -n <namespace> pod -o wide | grep ibm-block-csi`
+3. Check if the PersistentVolumeClaims (PVCs) are _Bound_.
+
+    `kubectl get pvc`
+
+    - If the PVCs are not in the _Bound_ state collect the events of all unbound PVCs. (See [Log collection for unbound PVCs](#log-collection-for-unbound-pvcs).)
+
+### Log collection for all CSI nodepods and their containers
+
+To collect logs for all nodepods, use the following commands:
+
+    nodepods=`kubectl get -n <namespace> pod -l app.kubernetes.io/component=csi-node --output=jsonpath={.items..metadata.name}`
+    
+    for pod in $nodepods; kubectl logs -n <namespace> $pod -c $container > /tmp/logs_collection/${pod}_${container}.log;done;done
 
-## Status collection for CSI pods, daemonset, and statefulset
 
-`kubectl get all -n <namespace>  -l csi`
+### Log collection for all CSI controller containers
 
-### Log collection for IBM block storage CSI driver controller
+To collect logs for all pods and containers, use the following commands:
+    
+    for container in kubectl get -n default pod ibm-block-csi-controller-0 -o jsonpath='{.spec.containers[*].name}';do kubectl logs -n default ibm-block-csi-controller-0 -c $container > /tmp/logs_collection/ibm-block-csi-controller-0_${container}.log;done
```

Comment:
1. why is it sometimes `<namespace>` and sometimes `default`?
2. sometimes `> logs/` and sometimes `/tmp/logs_collection/`?
3. is it important for this to be a one-liner? if not, we could reduce its length by predefining e.g. `podname = ibm-block-csi-controller-0`, perhaps even making it the same for both node and controller, thus reducing duplication as well

Reply:
1. it was a mistake on my part. I was supposed to change it. it is now changed in the new PR
2. updated
3. it is not important for it to be a one liner but node and controller are separate and so they can be separate commands. It's what I discussed with @roysahar . Roy - if you want to update it please let me know.
all changes can be found in  [PR 418](https://github.com/IBM/ibm-block-csi-driver/pull/418) for updates

Reply:
thanks!